src/connection.ts

```diff
@@ -168,6 +184,54 @@ export class Connection extends JSForceConnection {
     this.logger = this.tooling._logger = await Logger.child('connection');
   }
 
+  public async deploy(
+    zipInput: Buffer,
+    options: deployOptions,
+    callback?: Callback<AsyncResult>
+  ): Promise<DeployResultLocator<AsyncResult>> {
+    const rest = options.rest;
+    // neither API expects this option
+    delete options.rest;
+    if (rest) {
+      this.logger.debug('deploy with REST');
+      const headers = {
+        Authorization: this && `OAuth ${this.accessToken}`,
+        clientId: this.oauth2 && this.oauth2.clientId,
+        'Sforce-Call-Options': 'client=sfdx-core',
+      };
+      const url = `${this.baseUrl()}/metadata/deployRequest`;
+      const request = Transport.prototype._getHttpRequestModule();
+
+      return await new Promise((resolve, reject) => {
```

Comment:
Outside of try/catch blocks, `return await` is redundant.   You can just `return new Promise(...)`